build_runner/lib/src/changes/build_script_updates.dart

```diff
@@ -3,58 +3,140 @@
 // BSD-style license that can be found in the LICENSE file.
 
 import 'dart:async';
-import 'dart:io';
 import 'dart:mirrors';
 
 import 'package:build/build.dart';
-import 'package:build_runner/src/util/clock.dart';
 import 'package:logging/logging.dart';
-import 'package:path/path.dart' as path;
+import 'package:path/path.dart' as p;
 
 import '../asset/reader.dart';
+import '../asset_graph/graph.dart';
 import '../generate/options.dart';
 
 class BuildScriptUpdates {
-  final RunnerAssetReader _reader;
+  final AssetGraph _graph;
```

Comment:
can we get away with passing what is necessary as arguments to `hasBeenUpdated`?

Reply:
We could, but we would have to pass it through a few different functions - not sure what we would gain?